webui/react/src/utils/trial.ts

```diff
@@ -43,6 +43,26 @@ export const metricNameToValue = (metricName: MetricName): string => {
   return `${metricName.type}|${metricName.name}`;
 };
 
+export const metricNameFromValue = (metricValue: string): MetricName | undefined => {
+  const trainingPrefix = `${MetricType.Training}|`;
+  const validationPrefix = `${MetricType.Validation}|`;
+  if (metricValue.startsWith(trainingPrefix)) {
+    return {
+      name: metricValue.slice(trainingPrefix.length),
+      type: MetricType.Training,
+    };
+  } else if (metricValue.startsWith(validationPrefix)) {
+    return {
+      name: metricValue.slice(validationPrefix.length),
+      type: MetricType.Validation,
+    };
+  } else {
+    console.error("metricNameFromValue was called, but the metricName doesn't appear to " +
```

Comment:
ESLint rule says we shouldn't use the console. This case could occur if we add new types of metrics in the future but forget to update this function. How do we make sure we notice if that happens?

Reply:
we have a general `handleError` function in `ErrorHandler.ts`. One common way we've been using this is calling `handleError` with the `silent` object property to be `true` if `process.env.IS_DEV` is `true`. Meaning, if this error occurs while in the dev environment it will present it to the user via notification. Meanwhile in the background it will send it to segments as part of error analytics if telemetry has been enabled.